app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -17,6 +17,20 @@ class PipelineSampleReport extends React.Component {
     this.disableFilters = this.disableFilters.bind(this);
     this.enableFilters = this.enableFilters.bind(this);
   }
+  
+  componentDidMount() {
```

Comment:
Please find a way to do this without coponentDidMount().   It has been a major source of performance problems in the past.

Reply:
@boris-dimitrov, componentDidMount() is required to invoke a method after a component is mounted. the only thing that happens after the component is mounted is that the tags are hidden until the names are hovered on. Also when hovered on, it takes a maximum of 1millisecond to show and hide the tags. The other way to do this would be using DOM manipulation with jQuery which is not the React way.

Reply:
Please find a big report (>3000 rows) and measure the browser layout time before and after this change.   If there is a way to *not* introduced componentDidMount() please find it.   Literally every time it has been introduced on this page, performance has suffered, and I have had to delete it.

Reply:
hey @boris-dimitrov, I just did a test with huge data, it actually doesn't affect the performance cos it is not walking through the entire DOM, and is only called if a name is hovered on. in which case, the response is instantaneous.